### PR TITLE
*: Upgrade dependencies, cleanup unused, update prettier config

### PR DESCRIPTION
**Pull request by fordN**

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -2,5 +2,7 @@
   "packages": [
     "packages/*"
   ],
+  "npmClient": "yarn",
+  "useWorkspaces": true,
```
Comment (kaiwetlesen) [inline lerna.json:6]: Will this fix the issue with running `npm install`?

File: package.json [hunk 2]
```diff
@@ -25,15 +25,19 @@
...
     "@ethersproject/bignumber": "5.5.0",
     "@ethersproject/providers": "5.5.1",
-    "@urql/core": "2.3.6",
-    "@urql/exchange-execute": "1.2.2",
-    "scrypt": "https://registry.yarnpkg.com/@favware/skip-dependency/-/skip-dependency-1.0.2.tgz"
+    "@urql/core": "2.4.4",
+    "@urql/exchange-execute": "1.2.2"
   },
-  "dependencies": {
-    "scrypt": "https://registry.yarnpkg.com/@favware/skip-dependency/-/skip-dependency-1.0.2.tgz"
```
Comment (kaiwetlesen) [inline package.json:36]: Did we ever determine the reason for this dependency skip?

File: packages/indexer-agent/package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   "scripts": {
     "format": "prettier --write src/*.ts src/**/*.ts src/**/**/*.ts",
     "lint": "eslint . --ext .ts,.tsx",
-    "compile": "./node_modules/.bin/tsc",
+    "compile": "tsc",
```
Comment (kaiwetlesen) [inline packages/indexer-agent/package.json:13]: Will this work under all platforms wherein the path doesn't always get set up correctly?

File: packages/indexer-cli/package.json [hunk 4]
```diff
@@ -7,41 +7,41 @@
   "author": "The Graph Foundation",
   "license": "MIT",
   "scripts": {
-    "format": "prettier --write src/*.ts src/**/*.ts src/**/**/*.ts",
-    "lint": "eslint . --ext .ts,.tsx",
-    "compile": "./node_modules/.bin/tsc",
-    "prepare": "rm -rf dist && yarn format && yarn lint && yarn compile",
+    "format": "prettier --write 'src/**/*.ts'",
+    "lint": "eslint . --ext .ts,.tsx --fix",
```
Comment (kaiwetlesen) [inline packages/indexer-cli/package.json:11]: 🚀

File: packages/indexer-cli/src/commands/indexer/cost/get.ts [hunk 5]
```diff
@@ -50,15 +50,16 @@ module.exports = {
       : ['deployment', 'model', 'variables']
     const rawDeployment = ['model', 'variables'].includes(first) ? second : first
 
-    try{
-      if (rawDeployment !== 'all') {await validateDeploymentID(rawDeployment)}
+    try {
+      if (rawDeployment !== 'all') {
+        await validateDeploymentID(rawDeployment)
+      }
```
Comment (kaiwetlesen) [inline packages/indexer-cli/src/commands/indexer/cost/get.ts:56]: Guess this is the linter fix?

File: lerna.json (hunk 1 above)
Comment (fordN) [reply lerna.json:6]: Unfortunately doesn't fully fix it, but is part of my work to get there.  It's at least setting this mono repo up in a more explicit way.

File: package.json (hunk 2 above)
Comment (fordN) [reply package.json:36]: No, I couldn't find it being used anywhere so I simply removed.  I'm inclined to think that it was used for vector/connext payment channels before they were removed.

**Review by kaiwetlesen — APPROVED** [review]
Looks good! Guess there's just a couple questions we should have written down somewhere.

File: packages/indexer-agent/package.json (hunk 3 above)
Comment (fordN) [reply packages/indexer-agent/package.json:13]: Good question.  I don't have a definitive answer, but found that this makes it work with NPM whereas the node_modules/bin location execution we were using before did not.

File: packages/indexer-cli/src/commands/indexer/cost/get.ts (hunk 5 above)
Comment (fordN) [reply packages/indexer-cli/src/commands/indexer/cost/get.ts:56]: Yep!

File: lerna.json (hunk 1 above)
Comment (fordN) [reply lerna.json:6]: I need to do some more research on handling workspaces in `npm` and generally setting up a monorepo to work with both `npm` and `yarn.

**Comment by fordN** [comment]
Thanks for the quick review @kaiwetlesen! 🙏   
I'll merge when the CI builds and tests complete successfully.

**Pull request merged**